main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/Initializer.cs

```diff
@@ -119,7 +170,36 @@ async void OnFrameChanged (object s, EventArgs a)
 				disassemblyDoc.Select ();
 			}
 		}
-		
+
+		async System.Threading.Tasks.Task<Document> DownloadAndOpenFileAsync (StackFrame frame, int line, SourceLink sourceLink)
+		{
+			var pm = IdeApp.Workbench.ProgressMonitors.GetStatusProgressMonitor (
+				GettextCatalog.GetString ("Downloading ") + sourceLink.Uri,
+				Stock.StatusWorking,
+				true
+			);
+
+			Document doc = null;
+			try {
+				var downloadLocation = sourceLink.GetDownloadLocation (symbolCachePath);
+				Directory.GetParent (downloadLocation).Create ();
```

Comment:
```suggestion
				Directory.CreateDirectory (Path.GetDirectoryName (downloadLocation));
```

Avoids a redundant DirectoryInfo allocation.